k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -385,14 +352,19 @@ public void handleAsyncUntaggedResponse(ImapResponse response) {
                                 Timber.d("Got useful async untagged response: %s for %s", response, getLogId());
                             }
 
-                            idleStopper.stopIdle();
+                            synchronized (storedUntaggedResponses) {
+                                storedUntaggedResponses.add(response);
+                                if (!folder.getConnection().areMoreResponsesAvailable()) {
+                                    processStoredUntaggedResponses();
```

Comment:
@Valodim Sorry, I thought you were talking about the indentation. I have moved the processing command to the end of the method.